lib/view_component/compiler.rb

```diff
@@ -192,14 +192,23 @@ def templates
       @templates ||=
         begin
           extensions = ActionView::Template.template_handler_extensions
-
-          component_class._sidecar_files(extensions).each_with_object([]) do |path, memo|
-            pieces = File.basename(path).split(".")
-            memo << {
-              path: path,
-              variant: pieces.second.split("+").second&.to_sym,
-              handler: pieces.last
-            }
+          check_class = component_class
+
+          loop do
+            files = check_class._sidecar_files(extensions)
+            if files.length > 0
+              break files.map do |path|
```

Comment:
Instead of using `break` to return values, could we combine the above suggestion with `find`/`detect` to return the first value found?